### ISSUE TITLE
Failing to run benchmark

### ISSUE THREAD

**Issue by leitao** [PPC64/hhvm#1]
Currently hhvm for ppc64le (interpreted mode) is segfaulting to run the following benchmark:

http://www.php-benchmark-script.com/bench.zip

With the following error:

<pre>--------------------------------------
## |        PHP BENCHMARK SCRIPT        |

Start : 2015-06-17 13:12:49
Server : @
PHP version : 5.6.99-hhvm
## Platform : Linux

Core dumped: Segmentation fault
Stack trace in /tmp/stacktrace.16025.log
Segmentation fault (core dumped)
$ cat /tmp/stacktrace.16025.log

PHP Stacktrace:
#0  test_IfElse() called at [/home/ubuntu/hhvm/hhvm/hphp/hhvm/scrp/bench.php:74]

**Comment by leitao** [comment]
The stack trace is the following:

#0  0x000000001067b9c4 in HPHP::string_number_format(double, int, HPHP::String const&, HPHP::String const&) ()
#1  0x0000000011784da8 in HPHP::f_number_format(double, int, HPHP::Variant const&, HPHP::Variant const&) ()
#2  0x0000000012e87f84 in HPHP::Native::callFuncInt64Impl(HPHP::TypedValue\* (_)(HPHP::ActRec_), long_, int, double_, int) ()
#3  0x0000000010c14664 in void HPHP::Native::callFunc<true, false>(HPHP::Func const_, void_, HPHP::TypedValue_, int, HPHP::TypedValue&) ()
#4  0x0000000010e04380 in unsigned char_ HPHP::dispatchImpl<false>() ()
#5  0x0000000010e09290 in HPHP::enterVM(HPHP::ActRec_, HPHP::StackArgsState, HPHP::Resumable_, HPHP::ObjectData_, HPHP::VarEnv_) ()
#6  0x0000000012df6de4 in HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue_, HPHP::Func const_, HPHP::Variant const&, HPHP::ObjectData_, HPHP::Class_, HPHP::VarEnv_, HPHP::StringData_, HPHP::ExecutionContext::InvokeFlags) ()
#7  0x0000000012e5f5a0 in HPHP::ExecutionContext::invokeUnit(HPHP::TypedValue_, HPHP::Unit const_) ()
#8  0x000000001094dbb8 in HPHP::include_impl_invoke(HPHP::String const&, bool, char const_) ()
#9  0x00000000105fa614 in HPHP::hphp_invoke(HPHP::ExecutionContext_, std::string const&, bool, HPHP::Array const&, HPHP::VRefParamValue const&, std::string const&, std::string const&, bool&, std::string&, bool, bool, bool) ()
#10 0x00000000105fae74 in HPHP::hphp_invoke_simple(std::string const&, bool) ()
#11 0x000000001061e7d8 in HPHP::execute_program_impl(int, char**) ()
#12 0x00000000106219dc in HPHP::execute_program(int, char**) ()
#13 0x00000000105b53a0 in main ()

**Comment by lbianc** [comment]
Hi @leitao!

We have identified the problem and we just sent the fix.
We have disabled the SIMD support, since the port is in the beginning and some architecture specific code will be necessary for that. This solves the Segmentation Fault issue, but we are still analysing it, since there is no archictecture specific code around it.

The problem happens on a function cast for different type of data, at hphp/runtime/vm/native-func-caller.h:839. On that line there is a function call cast objects to int64_t, what is not working for PPC64.

We are still analysing this issue, but for now, the last couple of commits solves the problem.

Fell free to check it.

Thanks!
